### ISSUE TITLE
Challenge identifiers should be UUID's

### ISSUE THREAD

**Issue by zwade** [google/ctfscoreboard#5]
### k9delight: 

If challenge identifiers were UUID's, it would give us a variety of features, such as:

 - the ability to incrementally add levels and merge/restore them without likelyhood of conflict with existing levels
 - "level packs" where we can distribute the level descriptions afterwards, and people can add them to their own instance

### matir:

I completely agree, but more so, I think we should do the same for:

 - Categories (or possibly deterministic keys for merging)
 - Hints

Basically anything that's attached to the Challenges.

from: https://github.com/Matir/pwnableweb-scoreboard/issues/81
